gtda/images/filtrations.py

```diff
@@ -1036,3 +1038,227 @@ def plot(Xt, sample=0, colorscale='greys', origin='upper',
             title=f"Signed-distance filtration of image {sample}",
             plotly_params=plotly_params
             )
+
+
+@adapt_fit_transform_docs
+class DensityFiltration(BaseEstimator, TransformerMixin, PlotterMixin):
+    """Filtrations of 2D/3D binary images based on the number of neighboring
+    activated pixels.
+
+    The density filtration assigns to each pixel of a binary image a greyscale
+    value equal to the number of activated pixels with a ball centered around
+    it.
+
+    Parameters
+    ----------
+    radius : float, default: ``1``
+        The radius of the ball within which the number of activated pixels is
+        considered.
+
+    metric : string, or callable, optional, default: ``'euclidean'``
+        Determines a rule with which to calculate distances between
+        pairs of pixels.
+        If ``metric`` is a string, it must be one of the options allowed by
+        ``scipy.spatial.distance.pdist`` for its metric parameter, or a metric
+        listed in ``sklearn.pairwise.PAIRWISE_DISTANCE_FUNCTIONS``, including
+        "euclidean", "manhattan", or "cosine".
+        If ``metric`` is a callable function, it is called on each pair of
+        instances and the resulting value recorded. The callable should take
+        two arrays from the entry in `X` as input, and return a value
+        indicating the distance between them.
+
+    metric_params : dict, optional, default: ``{}``
+        Additional keyword arguments for the metric function.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    mask_ : ndarray of shape
```

Comment:
As a first-time reader and without reading the code, I didn't understand what this means. What shape does it have? How is it applied around each pixel?

Reply:
Indeed, those info are missing, thanks!